modules/system/rolls/wom-cast-test.js

```diff
@@ -112,31 +132,42 @@ export default class WomCastTest extends CastTest {
           break
       }
 
-      overcastData.usage[choice].count++;
+
+      // Subtract cost of overcasting from available SL
+      // AoE is separate column from target, so must be tested separately 
       if (choice == "target" && overcastData.usage.target.AoE) {
         overcastData.available = overcastData.available - overCastTable["AoE"][count].cost
-      } else {
+      } 
+      else if (choice == "other") {
+        overcastData.available = overcastData.available - 2
+      }
+      else {
         overcastData.available = overcastData.available - overCastTable[choice][count].cost
       }
 
+      overcastData.usage[choice].count++;
+
+      // Check if options are still available for more overcasting (now that count is incremented)
+      // It is not available if 1). no more entries in the table or 2). not enough available SL
       if (overcastData.usage.range) {
-        overcastData.usage.range.available = overcastData.available >= overCastTable.range[overcastData.usage.range.count].cost
+        overcastData.usage.range.available = overCastTable.range[overcastData.usage.range.count] && overcastData.available >= overCastTable.range[overcastData.usage.range.count].cost
       }
       if (overcastData.usage.target) {
         if (overcastData.usage.target.AoE) {
-          overcastData.usage.target.available = overcastData.available >= overCastTable.AoE[overcastData.usage.target.count].cost
+          overcastData.usage.target.available = overCastTable.AoE[overcastData.usage.target.count] && overcastData.available >= overCastTable.AoE[overcastData.usage.target.count].cost
         } else {          
-          overcastData.usage.target.available = overcastData.available >= overCastTable.target[overcastData.usage.target.count].cost
+          overcastData.usage.target.available = overCastTable.target[overcastData.usage.target.count] && overcastData.available >= overCastTable.target[overcastData.usage.target.count].cost
         }
       }
       if (overcastData.usage.duration) {
-        overcastData.usage.duration.available = overcastData.available >= overCastTable.duration[overcastData.usage.duration.count].cost
+        overcastData.usage.duration.available = overCastTable.duration[overcastData.usage.duration.count] && overcastData.available >= overCastTable.duration[overcastData.usage.duration.count].cost
       }
       if (overcastData.usage.damage) {
-        overcastData.usage.damage.available = overcastData.available >= overCastTable.damage[overcastData.usage.damage.count].cost
+        overcastData.usage.damage.available = overCastTable.damage[overcastData.usage.damage.count] && overcastData.available >= overCastTable.damage[overcastData.usage.damage.count].cost
       }
 
-      this.data.result.SL = `+${overcastData.available}`
+      // Subtract SL by the amount spent on overcasts
+      this.data.result.SL = `+${overcastData.originalSL - (overcastData.total - overcastData.available)}`
```

Comment:
i was not changing the original SL value, because i didnt see that its importent - however maybe it is, if for example this value is used in spell effects?

Reply:
Well, I saw two issues

- If you cast a CN 5 spell with 15 SL, you have 10 Overcasts
- If you spent 1 to +1 Damage, you now have 9
- Now, on the card, you went from 15 SL to 9 SL (should be 14)
- Then upon resetting, you went to 10 SL (should be 15)

As you say I don't think it's that important in any mechanics, but it's correct, and better to be safe

Reply:
Ok,. Now I see my mistake. I have been mostly checking this will full channeling so with CN = 0, and in this scenario issue doesn't appear